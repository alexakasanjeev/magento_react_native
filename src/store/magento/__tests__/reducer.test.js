import magentoReducer, { INITIAL_STATE } from '../reducer';
import { MAGENTO } from '../../../constants';

describe('magento reducer', () => {
  test('should return the initial state', () => {
    // Exercise
    const newState = magentoReducer(undefined, {});

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle init app loading', () => {
    // Setup
    const action = {
      type: MAGENTO.INIT_APP_LOADING,
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle init app success', () => {
    // Setup
    const action = {
      type: MAGENTO.INIT_APP_SUCCESS,
      payload: {
        storeConfig: [
          {
            id: 1,
          }
        ]
      }
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle init app failure', () => {
    // Setup
    const action = {
      type: MAGENTO.INIT_APP_FAILURE,
      payload: {
        errorMessage: 'Unable to initialize app.'
      }
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle currency loading', () => {
    // Setup
    const action = {
      type: MAGENTO.CURRENCY_LOADING,
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle currency success', () => {
    // Setup
    const action = {
      type: MAGENTO.CURRENCY_SUCCESS,
      payload: {
        currencyData: {
          available_currency_codes: ['INR', 'USD'],
          base_currency_code: 'USD',
          base_currency_symbol: '$',
          default_display_currency_code: 'INR',
          default_display_currency_symbol: 'â‚¹',
          exchange_rates: [],
        },
        displayCurrency: {
          code: 'USD',
          symbol: '$',
          rate: 1
        }
      }
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle currency failure', () => {
    // Setup
    const action = {
      type: MAGENTO.CURRENCY_FAILURE,
      payload: {
        errorMessage: 'Unable to fetch currency.'
      }
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle country loading', () => {
    // Setup
    const action = {
      type: MAGENTO.COUNTRIES_LOADING,
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle country success', () => {
    // Setup
    const action = {
      type: MAGENTO.COUNTRIES_SUCCESS,
      payload: {
        countries: [
          {
            id: 'IN'
          },
          {
            id: 'US'
          }
        ]
      }
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });

  test('should handle country failure', () => {
    // Setup
    const action = {
      type: MAGENTO.COUNTRIES_FAILURE,
      payload: {
        errorMessage: 'Unable to fetch country data.'
      }
    };

    // Exercise
    const newState = magentoReducer(INITIAL_STATE, action);

    // Verify
    expect(newState).toMatchSnapshot();
  });
});
