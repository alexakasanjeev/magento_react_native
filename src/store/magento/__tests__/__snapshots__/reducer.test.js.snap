// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`magento reducer should handle country failure 1`] = `
Object {
  "countryStatus": "ERROR",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "NOT_STARTED",
  "errorMessage": "Unable to fetch country data.",
  "storeConfigStatus": "NOT_STARTED",
}
`;

exports[`magento reducer should handle country loading 1`] = `
Object {
  "countryStatus": "LOADING",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "NOT_STARTED",
  "storeConfigStatus": "NOT_STARTED",
}
`;

exports[`magento reducer should handle country success 1`] = `
Object {
  "countries": Array [
    Object {
      "id": "IN",
    },
    Object {
      "id": "US",
    },
  ],
  "countryStatus": "SUCCESS",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "NOT_STARTED",
  "storeConfigStatus": "NOT_STARTED",
}
`;

exports[`magento reducer should handle currency failure 1`] = `
Object {
  "countryStatus": "NOT_STARTED",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "ERROR",
  "errorMessage": "Unable to fetch currency.",
  "storeConfigStatus": "NOT_STARTED",
}
`;

exports[`magento reducer should handle currency loading 1`] = `
Object {
  "countryStatus": "NOT_STARTED",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "LOADING",
  "storeConfigStatus": "NOT_STARTED",
}
`;

exports[`magento reducer should handle currency success 1`] = `
Object {
  "countryStatus": "NOT_STARTED",
  "currency": Object {
    "available_currency_codes": Array [
      "INR",
      "USD",
    ],
    "base_currency_code": "USD",
    "base_currency_symbol": "$",
    "default_display_currency_code": "INR",
    "default_display_currency_symbol": "â‚¹",
    "displayCurrencyCode": "USD",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "$",
    "exchange_rates": Array [],
  },
  "currencyStatus": "SUCCESS",
  "storeConfigStatus": "NOT_STARTED",
}
`;

exports[`magento reducer should handle init app failure 1`] = `
Object {
  "countryStatus": "NOT_STARTED",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "NOT_STARTED",
  "errorMessage": "Unable to initialize app.",
  "storeConfigStatus": "ERROR",
}
`;

exports[`magento reducer should handle init app loading 1`] = `
Object {
  "countryStatus": "NOT_STARTED",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "NOT_STARTED",
  "storeConfigStatus": "LOADING",
}
`;

exports[`magento reducer should handle init app success 1`] = `
Object {
  "countryStatus": "NOT_STARTED",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "NOT_STARTED",
  "storeConfig": Object {
    "id": 1,
  },
  "storeConfigStatus": "SUCCESS",
}
`;

exports[`magento reducer should return the initial state 1`] = `
Object {
  "countryStatus": "NOT_STARTED",
  "currency": Object {
    "default_display_currency_code": "",
    "default_display_currency_symbol": "",
    "displayCurrencyCode": "",
    "displayCurrencyExchangeRate": 1,
    "displayCurrencySymbol": "",
  },
  "currencyStatus": "NOT_STARTED",
  "storeConfigStatus": "NOT_STARTED",
}
`;
